application/helpers/expressions/em_cache_helper.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/**
+ * Discussion here: https://bugs.limesurvey.org/view.php?id=14859
+ *
+ * @since 2019-05-23
+ * @author Olle Haerstedt
+ */
+class EmCacheHelper
+{
+    /**
+     * Survey info from getSurveyInfo.
+     *
+     * @var array|null
+     */
+    protected $surveyinfo = null;
+
+    /**
+     * Set survey info used by this request.
+     *
+     * @param int $sid
+     * @return void
+     * @throws InvalidArgumentException if $surveyinfo is null.
+     */
+    public static function init(array $surveyinfo)
+    {
+        if (is_null($surveyinfo)) {
+            throw new \InvalidArgumentException('$surveyinfo is empty, cannot initialise helper');
+        }
+
+        self::$surveyinfo = $surveyinfo;
+    }
+
+    /**
+     * Flush cache. Should be done at all places where the cache is invalidated, e.g. at save survey/question/etc.
+     *
+     * @return void
+     * @throws EmCacheException if surveyinfo is not initialised.
+     */
+    public static function flush()
+    {
+        if (empty(self::$surveyinfo)) {
+            throw new EmCacheHelper('self::$surveyinfo is null, helper not initialised');
+        }
+
+        // Set survey cache array to empty.
+        \Yii::app()->emcache->set(self::$surveyinfo['sid'], []);
+    }
+
+    /**
+     * Flush ALL emcache, for all surveys.
+     *
+     * @return void
+     */
+    public static function flushAll()
+    {
+        \Yii::app()->emcache->flush();
+    }
+
+    /**
+     * Get cache value with $key.
+     *
+     * @param string $key
+     * @return mixed
+     */
+    public static function get($key)
+    {
+        if (!self::useCache()) {
+            return false;
+        }
+
+        $surveyCache = self::getSurveyCache();
+
+        if (empty($surveyCache)) {
+            return false;
+        }
+
+        if (empty($surveyCache[$key])) {
+            return false;
+        }
+
+        return $surveyCache[$key];
+    }
+
+    /**
+     * Set cache $value for $key.
+     *
+     * @param string $key
+     * @param mixed $value
+     * @return void
+     */
+    public static function set($key, $value)
+    {
+        if (!self::useCache()) {
+            return;
+        }
+
+        \Yii::app()->cache->set($key, $value);
```

Comment:
`\Yii::app()->emcache->set($key, $value);` if you want to allow different cache for em and for db

Reply:
Ok temporary solution for checking :)